File: haskell/merge.hs
merge xs [] = xs
merge [] ys = ys
merge (x:xs) (y:ys)
    | x < y = x:merge xs (y:ys)
    | otherwise = y:merge (x:xs) ys

merge' xs [] = xs
merge' [] ys = ys
merge' l1@(x:xs) l2@(y:ys)
    | x < y = x:merge' xs l2
    | otherwise = y:merge' l1 ys

File: python/merge.py
def merge(lista1, lista2):
	i1 = 0
	i2 = 0
	resultado = []
	while (i1 < len(lista1) and i2 < len(lista2)):
		if (lista1[i1] <= lista2[i2]):
			resultado.append(lista1[i1])
			i1=i1+1
		else:
			resultado.append(lista2[i2])
			i2=i2+1
	if i1 < len(lista1):
		resto = lista1[i1:]
	else:
		resto = lista2[i2:]

	return resultado + resto

	## return sorted (lista1 + lista2)
	
print(merge([1,2,3,4], [2,3,4,5]))
